src/info/worker.ts

```diff
@@ -0,0 +1,852 @@
+        if (rendering[i].entityType == "object") {
+          const name = rendering[i].name;
+          const centroid = rendering[i].centroid.map((x: any) => transformToVector(x));
+          const coords = rendering[i].contours;
+          const tCentroid = rendering[i].centroid.map((x: any) => transformPtToWorkspace(x))
+
+          let baseObj = {
+            name: name,
+            centroid: centroid,
+            coords: coords
+          }
+
+          let tObj = {
+            name: name,
+            centroid: tCentroid,
+            coords: coords
+          }
+
+          baseObjectData.push(baseObj);
+          objectData.push(tObj);
+        }
+
+        // find segments and map them to 2DIY workspace
+        if (rendering[i].entityType == "segment") {
+          const coords = rendering[i].contours.map((y: any) => y.map((x: any) => mapCoordsToVec(x.coordinates)));
+          const tCoords = rendering[i].contours.map((y: any) => y.map((x: any) => mapCoords(x.coordinates)));
+          let baseSeg = { coords: coords }
+          let tSeg = { coords: tCoords }
+          baseSegmentData.push(baseSeg);
+          segmentData.push(tSeg);
+        }
+
+        // find audio files
+        let audio = {
+          name: rendering[i].name,
+          offset: rendering[i].offset,
+          duration: rendering[i].duration,
+          entityType: rendering[i].entityType,
+          isStaticSegment: rendering[i].entityType.includes("static") ? true : false
+        }
+        audioData.push(audio);
+      }
+
+      objects = createObjs(objectData);
+      segments = createSegs(segmentData);
+
+      this.self.postMessage({
+        positions: { x: positions.x, y: positions.y },
+        objectData: createObjs(baseObjectData),
+        segmentData: createSegs(baseSegmentData),
+      });
+    }
+  }
+
+
+  /**
+   * Transforms base ML coordinate data for segments into array of segments.
+   * @param segmentInfo Array containing objects that contain segment coordinate data.
+   * @returns Array of segments.
+   */
+  function createSegs(segmentInfo: any): SubSegment[][] {
+    let data: SubSegment[][] = [];
+    for (const segs of segmentInfo) {
+      const segment: Array<SubSegment> = [];
+      const segmentCoords = segs.coords[0];
+      for (let i = 0; i < segmentCoords.length; i++) {
+        let coordinates = segmentCoords[i];
+        segment[i] = { coordinates };
+      }
+      data.push(segment);
+    }
+    return data;
+  }
+
+  /**
+   * Returns an array of objects in transformed 2DIY plane.
+   * @param objectData Array containing objects that contain object coordinate data. 
+   * @returns Array of objects.
+   */
+  function createObjs(objectData: any): SubSegment[][] {
+    let data: SubSegment[][] = [];
+
+    // loop through each object entity
+    for (const obj of objectData) {
+      const object: Array<SubSegment> = [];
+
+      // if ungrouped, just add the object directly
+      if (obj.centroid.length == 1) {
```

Comment:
the code will break when obj.centroid is undefined. may be you can add 
if (obj.centroid && obj.centroid.length == 1){}